PyPoE/poe/file/bundle.py

```diff
@@ -509,6 +528,27 @@ def _read(self, buffer: BytesIO):
                     directory_record.offset + directory_record.size
                 ]
             )
+        
+        self.hash_algorithm = None
+        if len(dirs := self.directories) > 0:
+            dir_iter = iter(dirs.values())
+            root_entry = next(dir_iter)
+            if root_entry.hash == 0x07e47507b4a92e53:
+                self.hash_algorithm = HASH_ALGORITHM.FNV1A64
+            else:
+                h = root_entry.hash
+                # Find seed from root directory hash via inverses
+                h ^= h >> 47
+                h = (h * 0x5F7A0EA7E59B19BD) % 2**64
+                h ^= h >> 47
+                for cand in dir_iter:
+                    if len(cand.paths) > 0:
+                        dir = cand.paths[0].rsplit('/', 1)[0]
```

Comment:
i think cand.path would work just fine, no need for rsplit on anything

Reply:
We need to split the paths as they are full paths to _files_ that all reside in the same directory, while we wish to validate against the full path to that common _directory_ itself. `rsplit` gets us two chunks, everything to the left of the last slash and the remainder that we discard.